MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/offline/CacheManagementActivity.java

```diff
@@ -0,0 +1,284 @@
+
+    // Fetch the latest Remote Config values from Firebase and then adjust the cache if needed.
+    firebaseRemoteConfig.fetchAndActivate().addOnCompleteListener(this,
+        new OnCompleteListener<Boolean>() {
+          @Override
+          public void onComplete(@NonNull Task<Boolean> task) {
+            if (task.isSuccessful()) {
+              adjustCacheBasedOnRemoteConfigServerSettings();
+            } else {
+              Logger.d("CacheManagementActivity", "Couldn\'t get Firebase cache "
+                  + "management remote config settings from Firebase.");
+            }
+          }
+        });
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            findViewById(R.id.invalidate_ambient_cache_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                invalidateAmbientCache();
+              }
+            });
+
+            findViewById(R.id.set_max_ambient_cache_size_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                EditText newMaxAmbientCacheSizeTextView = findViewById(
+                    R.id.set_max_ambient_cache_size_button_editText);
+                long newMaxAmbientCacheSize = Long.valueOf(
+                    newMaxAmbientCacheSizeTextView.getText().toString());
+                setNewMaxAmbientCache(newMaxAmbientCacheSize);
+              }
+            });
+
+            findViewById(R.id.clear_ambient_cache_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                clearAmbientCache();
+              }
+            });
+
+            findViewById(R.id.reset_database_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                resetDatabase();
+              }
+            });
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Invalidate the ambient cache. See ANDROID-DOCS-URL-HERE for more
+   * information about this method.
+   */
+  private void invalidateAmbientCache() {
+    OfflineManager fileSource = OfflineManager.getInstance(CacheManagementActivity.this);
+    fileSource.invalidateAmbientCache(new OfflineManager.FileSourceCallback() {
+      @Override
+      public void onSuccess() {
+        showToast(R.string.cache_invalidated_toast_confirmation);
+      }
+
+      @Override
+      public void onError(@NonNull String message) {
+        showToast(R.string.cache_invalidated_toast_error);
```

Comment:
Would be nice to pass the error message tom the toast as well.
Same for below occurrences.

Reply:
Done